connectors/src/connectors/microsoft/temporal/workflows.ts

```diff
@@ -121,3 +135,22 @@ export function microsoftFullSyncSitesWorkflowId(connectorId: ModelId) {
 export function microsoftIncrementalSyncWorkflowId(connectorId: ModelId) {
   return `microsoft-incrementalSync-${connectorId}`;
 }
+
+export function microsoftDeletionWorkflowId(
+  connectorId: ModelId,
+  nodeIdsToDelete: string[]
+) {
+  // Simple, deterministic hashing function
+  function simpleHash(input: string): string {
```

Comment:
@philipperolet had to implement this simple function as Temporal doesn't let you import non deterministic libraries (like crypto or blake3)

Reply:
hmm creating a custom hash for this task is overkill imo

my initial suggestion of a hash may retrospectively had been suboptimal; actually just concatenating the last 4 chars of every internalId, capped at 256, seems more readable
apologies for the bad idea --- and don't hesitate to challenge my comments, they may need it :wink: